scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2734,12 +2748,16 @@ def start(self):
             )
 
     def stop(self, wait_on_join=True, join_timeout=5):
+        self.__stopped = True
+
         if self.__controlled_warmer is not None:
             self.__controlled_warmer.stop(
                 wait_on_join=wait_on_join, join_timeout=join_timeout
             )
 
-        self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
+        if self.__thread.isAlive():
```

Comment:
This is needed because threads might not have been started yet when stop is called.

```bash
2020-10-02 14:00:26.258Z ERROR [core] [scalyr_logging.py:618] Failed to stop the metric log due to an exception :stack_trace:
  Traceback (most recent call last):
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/monitors_manager.py", line 201, in stop_manager
      monitor.stop(join_timeout=max_wait)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 4195, in stop
      self.__container_checker.stop(wait_on_join, join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 2746, in stop
      self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1226, in stop
      self.join(join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1248, in join
      threading.Thread.join(self, timeout)
    File "/home/kami/.pythonz/pythons/CPython-3.6.9/lib/python3.6/threading.py", line 1051, in join
      raise RuntimeError("cannot join thread before it is started")
  RuntimeError: cannot join thread before it is started
```